File: mobile/lib/features/tasks/domain/entities/task.dart
import 'package:equatable/equatable.dart';
import 'task_destination.dart';
import 'warehouse.dart';

enum TaskStatus { pending, in_progress, done, cancelled }

class Task extends Equatable {
  final String id;
  final String salesId;
  final String? customerId;
  final String? customerName;
  final String? warehouseId;
  final Warehouse? warehouse;
  final String title;
  final String description;
  final List<TaskDestination> destinations;
  final TaskStatus status;
  final DateTime? dueDate;
  final DateTime? completedAt;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Task({
    required this.id,
    required this.salesId,
    this.customerId,
    this.customerName,
    this.warehouseId,
    this.warehouse,
    required this.title,
    required this.description,
    this.destinations = const [],
    required this.status,
    this.dueDate,
    this.completedAt,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        salesId,
        customerId,
        customerName,
        warehouseId,
        warehouse,
        title,
        description,
        destinations,
        status,
        dueDate,
        completedAt,
        createdAt,
        updatedAt,
      ];
}

File: backend/internal/domain/models/task.go
package models

import (
	"crm_wowin_backend/pkg/utils"

	"github.com/google/uuid"
)

type TaskStatus string

const (
	TaskStatusTodo       TaskStatus = "pending"
	TaskStatusInProgress TaskStatus = "in_progress"
	TaskStatusCompleted  TaskStatus = "done"
	TaskStatusCancelled  TaskStatus = "cancelled"
)

type Task struct {
	ID           uuid.UUID         `json:"id"`
	SalesID      uuid.UUID         `json:"sales_id"`
	WarehouseID  uuid.UUID         `json:"warehouse_id"` // Mandatory as per user feedback
	Warehouse    *Warehouse        `json:"warehouse,omitempty"`
	Title        string            `json:"title"`
	Description  string            `json:"description"`
	Destinations []TaskDestination `json:"destinations,omitempty"`
	Status       TaskStatus        `json:"status"`
	DueDate      *utils.FlexTime   `json:"due_date,omitempty"`
	CompletedAt  *utils.FlexTime   `json:"completed_at,omitempty"`
	CreatedAt    utils.FlexTime    `json:"created_at"`
	UpdatedAt    utils.FlexTime    `json:"updated_at"`
}
